cpu/esp8266/Makefile.dep

```diff
@@ -0,0 +1,96 @@
+# additional modules dependencies
```

Comment:
This file I don't understand. Why are there cpu-specific dependencies?
For ```newlib_syscalls_default```, is newlib usually not built, or without syscalls?
For ndn-riot, why does it have additional dependencies for esp8266?

Reply:
I thought it is the right way to place all dependencies in ```cpu/esp8266/Makefile.dep``` that are independent of the used board. There were other examples that are doing it in that way.
```
cpu/efm32/Makefile.dep
cpu/kinetis/Makefile.dep
cpu/stm32_common/Makefile.dep
cpu/fe310/Makefile.dep
```

Using module ```newlib_syscalls_default``` was necessary since ```esp-open-sdk``` newlib is missing a number of functions.

It seems that ```ndn-riot``` is using cryptographic functions. Without using modules ```crypto``` and ```cipher_modes``` I got undefined references:
```
/home/gs/src/RIOT-Xtensa-ESP.working/examples/ndn-ping/bin/esp8266-esp-12x/ndn-encoding.a(data.o): In function `ndn_data_create2':
/home/gs/src/RIOT-Xtensa-ESP.working/examples/ndn-ping/bin/pkg/esp8266-esp-12x/ndn-riot/encoding/data.c:345: warning: undefined reference to `CIPHER_AES_128'
/home/gs/src/RIOT-Xtensa-ESP.working/examples/ndn-ping/bin/pkg/esp8266-esp-12x/ndn-riot/encoding/data.c:347: warning: undefined reference to `cipher_init'
/home/gs/src/RIOT-Xtensa-ESP.working/examples/ndn-ping/bin/pkg/esp8266-esp-12x/ndn-riot/encoding/data.c:347: warning: undefined reference to `cipher_encrypt_ccm'
/home/gs/src/RIOT-Xtensa-ESP.working/examples/ndn-ping/bin/pkg/esp8266-esp-12x/ndn-riot/encoding/data.c:361: warning: undefined reference to `cipher_init'
/home/gs/src/RIOT-Xtensa-ESP.working/examples/ndn-ping/bin/pkg/esp8266-esp-12x/ndn-riot/encoding/data.c:425: warning: undefined reference to `cipher_encrypt_ccm'
/home/gs/src/RIOT-Xtensa-ESP.working/examples/ndn-ping/bin/esp8266-esp-12x/ndn-encoding.a(data.o): In function `ndn_data_verify_signature':
/home/gs/src/RIOT-Xtensa-ESP.working/examples/ndn-ping/bin/pkg/esp8266-esp-12x/ndn-riot/encoding/data.c:860: warning: undefined reference to `cipher_decrypt_ccm'
/home/gs/src/RIOT-Xtensa-ESP.working/examples/ndn-ping/bin/pkg/esp8266-esp-12x/ndn-riot/encoding/data.c:905: warning: undefined reference to `cipher_init'
/home/gs/src/RIOT-Xtensa-ESP.working/examples/ndn-ping/bin/pkg/esp8266-esp-12x/ndn-riot/encoding/data.c:921: warning: undefined reference to `cipher_decrypt_ccm'
```
Maybe this is dependency that should be handled by the example ```ndn-riot```. I don't know.

Reply:
@kasper30 
I used riotdocker to investigated the dependency of ```ndn-riot``` and the ```crypto```/```cipher_modes``` modules a bit more.

Several functions in ```ndn-riot/encoding/data.c``` require cryptographic functionality. However, ```example/ndn-ping``` seems not to use these functions from ```ndn-riot```. Even though the archive ```encoding.a``` contains some undefined symbols to the cryptographic functions, the linker on platforms like AVR doesn't try to to resolve these symbols. That means that these symbols are not really used.

The GNU linker on Xtensa platform seems to be more restrict and tries to resolve these undefined symbols even though they are note really needed. I already added a linker option ```--warn-unresolved-symbols``` so that the compilation doesn't fail and throws only warnings in such case. Only using the ```crypto```/```cipher_modes``` could solve this problem.

Another option would be to add the linker option ```--unresolved-symbols=ignore-all```. IMHO, this would be definitely the wrong way.

The question for me now is the following. If some functions of ```ndn-riot``` module require cryptographic functionality, should not the makefile of that module resolve the dependencies?